httpbp/server.go

```diff
@@ -0,0 +1,271 @@
+package httpbp
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+
+	baseplate "github.com/reddit/baseplate.go"
+	"github.com/reddit/baseplate.go/batcherror"
+)
+
+// EndpointRegistry is the minimal interface needed by a Baseplate HTTP server for
+// the underlying HTTP server.
+//
+// *http.ServeMux implements this interface and is the default EndpointRegistry
+// used by NewBaseplateServer.
+type EndpointRegistry interface {
+	http.Handler
+
+	Handle(pattern string, handler http.Handler)
+}
+
+var (
+	_ EndpointRegistry = (*http.ServeMux)(nil)
+)
+
+type httpHandlerFactory struct {
+	middlewares []Middleware
+}
+
+func (f httpHandlerFactory) NewHandler(name string, handle HandlerFunc, middlewares ...Middleware) http.Handler {
+	wrappers := make([]Middleware, 0, len(f.middlewares)+len(middlewares))
+	wrappers = append(wrappers, f.middlewares...)
+	wrappers = append(wrappers, middlewares...)
+	return NewHandler(name, handle, wrappers...)
+}
+
+// Pattern is the pattern passed to a EndpointRegistry when registering an
+// Endpoint.
+type Pattern string
```

Comment:
I wanted to say something like `Path` or `EndpointPath` would be better, but `ServerMux` also calls it pattern :man_shrugging: